salt/utils/versions.py

```diff
@@ -18,18 +18,29 @@
 import sys
 import warnings
 
-from setuptools._distutils.version import LooseVersion as _LooseVersion
-from setuptools._distutils.version import StrictVersion as _StrictVersion
+log = logging.getLogger(__name__)
+
+try:
+    from setuptools.distutils.version import LooseVersion as _LooseVersion
+    from setuptools.distutils.version import StrictVersion as _StrictVersion
+except ImportError:
+    log.debug(
+        "unable to import from setuptools.distutils.version, will attempt setuptools._distutils.version"
+    )
+    try:
+        from setuptools._distutils.version import LooseVersion as _LooseVersion
+        from setuptools._distutils.version import StrictVersion as _StrictVersion
+    except:
+        log.debug("unable to import from setuptools._distutils.version")
+        raise ImportError()
+
 
 # pylint: enable=blacklisted-module
 import salt.version
 
 # pylint: disable=blacklisted-module
```

Comment:
```suggestion
# pylint: disable=blacklisted-module
try:
    from setuptools._distutils.version import LooseVersion as _LooseVersion
    from setuptools._distutils.version import StrictVersion as _StrictVersion
except ImportError:
    try:
        from setuptools.distutils.version import LooseVersion as _LooseVersion
        from setuptools.distutils.version import StrictVersion as _StrictVersion
    except ImportError:
        from distutils.version import LooseVersion as _LooseVersion
        from distutils.version import StrictVersion as _StrictVersion

# pylint: enable=blacklisted-module
import salt.version

log = logging.getLogger(__name__)
```

Reply:
Just updated and submitted, can you look at recent code

Reply:
> I believe logging is important

In this particular scenario, will fallbacks in place, it's not.

Reply:
okay logging is gone, and implemented as suggested